http-server/src/server.rs

```diff
@@ -0,0 +1,311 @@
+	pub async fn new(url: impl AsRef<str>, config: HttpConfig) -> Result<Self, Box<dyn error::Error + Send + Sync>> {
+		let sockaddr = url.as_ref().parse()?;
+		let transport_server = HttpTransportServer::new(&sockaddr, config).await?;
+		let local_addr = *transport_server.local_addr();
+
+		// We use an unbounded channel because the only exchanged messages concern registering
+		// methods. The volume of messages is therefore very low and it doesn't make sense to have
+		// a backpressure mechanism.
+		// TODO: that's not true anymore ^
+		let (to_back, from_front) = mpsc::unbounded();
+
+		async_std::task::spawn(async move {
+			background_task(transport_server.into(), from_front).await;
+		});
+
+		Ok(Server {
+			local_addr,
+			to_back,
+			registered_methods: Arc::new(Mutex::new(HashSet::new())),
+			next_subscription_unique_id: Arc::new(atomic::AtomicUsize::new(0)),
+		})
+	}
+
+	/// Local socket address of the transport server.
+	pub fn local_addr(&self) -> &SocketAddr {
+		&self.local_addr
+	}
+
+	/// Registers a notification method name towards the server.
+	///
+	/// Clients will then be able to call this method.
+	/// The returned object allows you to process incoming notifications.
+	///
+	/// If `allow_losses` is true, then the server is allowed to drop notifications if the
+	/// notifications handler (i.e. the code that uses [`RegisteredNotifications`]) is too slow
+	/// to process notifications.
+	///
+	/// Returns an error if the method name was already registered.
+	pub fn register_notification(
+		&self,
+		method_name: String,
+		allow_losses: bool,
+	) -> Result<RegisteredNotification, Error> {
+		if !self.registered_methods.lock().insert(method_name.clone()) {
+			return Err(Error::MethodAlreadyRegistered(method_name));
+		}
+
+		log::trace!("[frontend]: register_notification={}", method_name);
+		let (tx, rx) = mpsc::channel(32);
+
+		self.to_back
+			.unbounded_send(FrontToBack::RegisterNotifications { name: method_name, handler: tx, allow_losses })
+			.map_err(|e| Error::Internal(e.into_send_error()))?;
+
+		Ok(RegisteredNotification { queries_rx: rx })
+	}
+
+	/// Registers a method towards the server.
+	///
+	/// Clients will then be able to call this method.
+	/// The returned object allows you to handle incoming requests.
+	///
+	/// Contrary to [`register_notifications`](Server::register_notifications), there is no
+	/// `allow_losses` parameter here. If the handler is too slow to process requests, then the
+	/// server automatically returns an "internal error" to the client.
+	///
+	/// Returns an error if the method name was already registered.
+	pub fn register_method(&self, method_name: String) -> Result<RegisteredMethod, Error> {
+		if !self.registered_methods.lock().insert(method_name.clone()) {
+			return Err(Error::MethodAlreadyRegistered(method_name));
+		}
+
+		log::trace!("[frontend]: register_method={}", method_name);
+		let (tx, rx) = mpsc::channel(32);
```

Comment:
Whether similar parameters should be configurable, I think this kind of parameters may affect performance and availability